vulnerabilities/importers/apache_httpd.py

```diff
@@ -81,56 +87,82 @@ def to_advisory(self, data):
+        fixed_versions = []
+        for timeline_object in data["timeline"]:
+            timeline_value = timeline_object["value"]
+            if "release" in timeline_value:
+                split_timeline_value = timeline_value.split(" ")
+                if "never" in timeline_value:
+                    continue
+                if "release" in split_timeline_value[-1]:
+                    fixed_versions.append(split_timeline_value[0])
+                if "release" in split_timeline_value[0]:
+                    fixed_versions.append(split_timeline_value[-1])
 
-        affected_packages = []
-        fixed_packages = []
-
-        for version_range in fixed_version_ranges:
-            fixed_packages.extend(
-                [
-                    PackageURL(type="apache", name="httpd", version=version)
-                    for version in self.version_api.get("apache/httpd").valid_versions
-                    if SemverVersion(version) in version_range
-                ]
-            )
+        affected_version_range = self.to_version_ranges(versions_data, fixed_versions)
 
-        for version_range in affected_version_ranges:
-            affected_packages.extend(
-                [
-                    PackageURL(type="apache", name="httpd", version=version)
-                    for version in self.version_api.get("apache/httpd").valid_versions
-                    if SemverVersion(version) in version_range
-                ]
+        affected_packages = []
+        if affected_version_range:
+            affected_packages.append(
+                AffectedPackage(
+                    package=PackageURL(
+                        type="generic",
+                        name="apache_httpd",
+                    ),
+                    affected_version_range=affected_version_range,
+                )
             )
 
         return AdvisoryData(
-            vulnerability_id=cve,
+            aliases=[alias],
             summary=description,
-            affected_packages=nearest_patched_package(affected_packages, fixed_packages),
+            affected_packages=affected_packages,
             references=[reference],
         )
 
-    def to_version_ranges(self, versions_data):
-        fixed_version_ranges = []
-        affected_version_ranges = []
+    def to_version_ranges(self, versions_data, fixed_versions):
+        constraints = []
         for version_data in versions_data:
             version_value = version_data["version_value"]
             range_expression = version_data["version_affected"]
-            if range_expression == "<":
-                fixed_version_ranges.append(
-                    VersionRange.from_scheme_version_spec_string(
-                        "semver", ">={}".format(version_value)
+            if range_expression not in KNOWN_RANGE_EXPRESSIONS:
+                raise ValueError(f"unknown comparator found! {range_expression}")
+            if range_expression == ">=" or range_expression == "!<":
```

Comment:
IMHO you could instead use something like this:
```Python
comparator_by_range_expression = {
    ">=": ">=",
    "!<": ">=",
    "<=": "<=",
    "=": "=",
}
comparator = comparator_by_range_expression.get(range_expression)
if comparator:
    constraints.append(
        VersionConstraint(comparator=comparator, version=SemverVersion(version_value))
    )    
```
This less duplicated code to express the same thing

Reply:
An elegant simplification!  Done.